pkg/controllers/nodeclass/validation.go

```diff
@@ -267,7 +267,7 @@ func (v *Validation) validateRunInstancesAuthorization(
 	}); awserrors.IgnoreDryRunError(err) != nil {
 		// If we get InstanceProfile NotFound, but we have a resolved instance profile in the status,
 		// this means there is most likely an eventual consistency issue and we just need to requeue
-		if awserrors.IsInstanceProfileNotFound(err) || awserrors.IsRateLimitedError(err) {
+		if (awserrors.IsInstanceProfileNotFound(err) && nodeClass.Spec.InstanceProfile == nil) || awserrors.IsRateLimitedError(err) {
```

Comment:
If we are trying to launch an instance, but are referencing an instance profile that doesn't exist, I wonder if we want to surface this in a different way other than raising an auth error in the validation controller -- realistically, this is something that I would expect the instance profile reconciler to catch rather than catching it in an auth check. Thoughts?